pkg/kubelet/kubelet_pods.go

```diff
@@ -1105,8 +1105,8 @@ func (kl *Kubelet) HandlePodCleanups() error {
 	}
 
 	// Stop probing pods that are not running
-	klog.V(3).InfoS("Clean up probes for terminating and terminated pods")
-	kl.probeManager.CleanupPods(runningPods)
+	klog.V(3).InfoS("Clean up probes for terminated pods")
+	kl.probeManager.CleanupPods(possiblyRunningPods)
```

Comment:
just for my own understanding - `possiblyRunningPods` refers to pods which are in progress of terminating right? Why do we want need to stop probing here in addition to logic in `syncPod`?

Reply:
HandlePodCleanups() is a goroutine which is running periodically to make sure pods are cleaned up. In my testing, 'runningPods' does not include terminating pods. This would cause the probes to be terminated prematurely (due to the natural race condition between the termination pod_worker logic and HandlePodCleanups).

I didn't want to change the logic of removing the probeManager call from HandlePodCleanups due to unknown side effects.